python_modules/dagster/dagster/components/utils.py

```diff
@@ -183,3 +183,45 @@ def get_path_from_module(module: ModuleType) -> Path:
         else None
     )
     return check.not_none(module_path, f"Module {module.__name__} has no filepath")
+
+
+def copy_fields_to_model(copy_from: type[BaseModel], copy_to: type[BaseModel]) -> type[BaseModel]:
+    """Given two models, creates a copy of the second model with the fields of the first model.
+
+    Args:
+        copy_from (type[BaseModel]): The model to copy the fields from.
+        copy_to (type[BaseModel]): The model to copy the fields to.
+
+    Returns:
+        type[BaseModel]: A new model which subclasses the `copy_to` model and has
+        the fields of the `copy_from` model.
+
+    Example:
+        ```python
+            class MyModel(BaseModel):
+                field1: int = 5
+                field2: str = "hello"
+
+            class MyBaseModel(BaseModel):
+                ...
+
+            class MyModelWithMyBaseModel(copy_fields_to_model(MyModel, MyBaseModel)):
+                ...
+
+            assert isinstance(MyModelWithMyBaseModel, MyBaseModel)
+            assert MyModelWithMyBaseModel.field1 == 5
+            assert MyModelWithMyBaseModel.field2 == "hello"
+    ```
+
+    """
+    field_definitions: dict[str, tuple[type, Any]] = {
+        field_name: (cast("type", field.annotation), field)
+        for field_name, field in copy_from.model_fields.items()
+    }
+
+    return create_model(
+        f"{copy_to.__name__}With{copy_from.__name__}",
+        __base__=copy_to,
+        __doc__=copy_to.__doc__,
+        **field_definitions,  # type: ignore
+    )
```

Comment:
grizzly - whats the backstory on this? no better way?

Reply:
#30765 introduced the Snowflake component - we want it to be configurable similar to the `SnowflakeResource` in YAML but there was hesitancy to subclass it directly. The other option would be to have the connection component "hold" a resource, e.g.

```yaml
type: dagster_snowflake.SnowflakeConnectionComponent

attribute:
  connection:
    user: ...
```

But this feels a bit unergonomic - this util lets us just copy those fields to the component itself. Agree this is gross and open to other suggestions.

Reply:
would it work to refactor the existing resources to inherit `ConfigruableResource` and a Mixin class that could get inherited here as well?